datarepo-clienttests/src/main/java/scripts/testscripts/CreateSnapshot.java

```diff
@@ -0,0 +1,199 @@
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.config.TestUserSpecification;
+import scripts.testscripts.baseclasses.SimpleDataset;
+import scripts.utils.DataRepoUtils;
+
+public class CreateSnapshot extends SimpleDataset {
+  private static final Logger logger = LoggerFactory.getLogger(CreateSnapshot.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public CreateSnapshot() {
+    super();
+  }
+
+  private SnapshotModel snapshotModel;
+
+  private static List<BlobId> scratchFiles = new ArrayList<>();
+
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    // create the profile and dataset
+    super.setup(testUsers);
+
+    // get the ApiClient for the snapshot creator, same as the dataset creator
+    ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
+    RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+
+    // load data into the new dataset
+    // note that there's a fileref in the dataset
+    // ingest a file -- TODO CannedTestData.getMeA1KBFile
+    URI sourceUri = new URI("gs://jade-testdata/fileloadprofiletest/1KBfile.txt");
+
+    String loadTag = FileUtils.randomizeName("lookupTest");
+    BulkLoadArrayRequestModel fileLoadModelArray =
+        new BulkLoadArrayRequestModel()
+            .profileId(datasetSummaryModel.getDefaultProfileId())
+            .loadTag(loadTag)
+            .maxFailedFileLoads(0);
+
+    List<String> fileIds = new ArrayList<>();
+    for (int i = 0; i < filesToLoad; i++) {
+      String targetPath = "/testrunner/IngestFile/" + FileUtils.randomizeName("") + ".txt";
+
+      BulkLoadFileModel fileLoadModel =
+          new BulkLoadFileModel()
+              .sourcePath(sourceUri.toString())
+              .description("IngestFile")
+              .mimeType("text/plain")
+              .targetPath(targetPath);
+      fileLoadModelArray.addLoadArrayItem(fileLoadModel);
+      if (fileLoadModelArray.getLoadArray().size() % 1000 == 0 || i == filesToLoad - 1) {
+        JobModel ingestFileJobResponse =
+            repositoryApi.bulkFileLoadArray(datasetSummaryModel.getId(), fileLoadModelArray);
+        ingestFileJobResponse =
+            DataRepoUtils.waitForJobToFinish(repositoryApi, ingestFileJobResponse);
+        BulkLoadArrayResultModel bulkLoadArrayResultModel =
+            DataRepoUtils.expectJobSuccess(
+                repositoryApi, ingestFileJobResponse, BulkLoadArrayResultModel.class);
+
+        try (Stream<BulkLoadFileResultModel> resultStream =
+            bulkLoadArrayResultModel.getLoadFileResults().stream()) {
+          fileIds.addAll(
+              resultStream.map(BulkLoadFileResultModel::getFileId).collect(Collectors.toList()));
```

Comment:
You can avoid creating a collection here by doing an `add` in the streaming operation
```suggestion
            resultStream.map(BulkLoadFileResultModel::getFileId).forEach(fileIds::add);
```